docs/dice.md

```diff
@@ -0,0 +1,170 @@
+  codeDescriptor          [1] EXPLICIT OCTET STRING OPTIONAL,
+  configurationHash       [2] EXPLICIT OCTET STRING OPTIONAL,
+  configurationDescriptor [3] EXPLICIT OCTET STRING OPTIONAL,
+  authorityHash           [4] EXPLICIT OCTET STRING OPTIONAL,
+  authorityDescriptor     [5] EXPLICIT OCTET STRING OPTIONAL,
+  mode                    [6] EXPLICIT Mode OPTIONAL,
+}
+```
+
+For further details regarding the Open DICE X.509 profile, see the [Certificate Details](https://pigweed.googlesource.com/open-dice/+/HEAD/docs/specification.md#certificate-details) section of the spec.
+
+## DICE from a Veraison's Perspective
+
+Miscellaneous implementation & deployment considerations.
+
+### Asymmetric Crypto Identities
+
+Typically, DICE identities are associated with asymmetric key pairs.
+
+> NOTE: This mode is, from a security perspective, the most desirable.  In fact, when using public keys for verification, the verifier only needs to take care of their availability and integrity.
+
+Effectively a verifier only stores the device trust anchor, which could be an external root CA -- for example the ODM's -- or the self-signed certificate of the device in case there is no external CA to anchor the cert chain.  The former providing a much more scalable solution, with individual device revocation handled via standard CRLs.
+
+### Symmetric Crypto Identities
+
+"Symmetric Identity Based Device Attestation" describes how to associate DICE identities with symmetric crypto keys.
+
+In this case, the verifier must have a-priori knowledge of the device secret as well as a detailed understanding of the identity derivation algorithm in order to verify the device's authenticated assertions.  This assumes a level of coupling between attesters and verifiers that, I reckon, is not achievable using only the available TCG standards, which tend to be quite liberal around the implementation details.  The Open-DICE specification only allows identities based on asymmetric keys.
+
+> NOTE: This mode shifts all the costs on the verification infra.  It is an operational nightmare with a vast attack surface and very limited scalability.  It should be deemed acceptable only when the cost of a key compromise / disclosure is (very) low, and the device fleet is roughly static with very infrequent software updates.
+
+### Verification
+
+We expect that, typically, DICE evidence submitted for verification is not bound to an explicit challenge-response session.  This is likely to require defining a new Veraison API alongside the existing [challenge-response](api/challenge-response/README.md) endpoint.
+
+Given the fact that TCG DICE has a few loose ends, mainly in terms of evidence encoding formats and the nature of cryptographic identities, Veraison will have to be as flexible as possible while clearly documenting the supported feature set.
+
+Open-DICE defines a stricter (and smaller) set of requirements and therefore it'd be probably easier to implement fully.
+
```

Comment:
Is it worth noting here that currently neither spec discusses on-wire format / serialisation & that adds uncertainty to the verification process?

Reply:
Are you thinking of the lack of media types for transporting DICE material?

Reply:
That's a specific, but I was generally thinking of this as a challenge for verification that the specs don't address, hence leaving the prospect for an (open/TCG)) DICE deployment based on Veraison as potentially needing to add some custom handling at the front end rather than being a full reference

Reply:
The open-DICE spec is much more crisp / easy to implement than TCG DICE.  I think the only real problem there is the potential for hybrid CWT/x.509 chains.

TCG DICE on the contrary will necessarily need some hammering to be fully supported.  But this point  is articulated already in L156.  Or do you think that is not sufficient?

Reply:
Thinking about it some more, that probably covers it, I may be just being over sensitive to what a consuming parties expectations might be, but we have time to accommodate such conversations.